### PR TITLE
Added omp requirement to build on Linux on README

### LINKED ISSUE #3740: Missing requirement in building guidelines for Linux
Using the given commands I am not able to build on Ubuntu 18.04 because of missing libomp-dev.

### PR DESCRIPTION
**Pull request by artidoro**
Fixes #3740.

Adds requirement and sample command to build on Linux.

### CODE REVIEW

File: docs/building/unix-instructions.md [hunk 1]
```diff
@@ -35,6 +36,7 @@ For example, for Ubuntu 16.x:
 sudo apt-get update
 sudo apt-get install git clang-3.9 cmake libunwind8 curl
 sudo apt-get install libssl1.0.0 libicu55
+sudo apt-get install libomp-dev
```
Comment (codemzs) [inline docs/building/unix-instructions.md]: Is this specific to all distros of Linux or just Ubuntu 18.04? If it is specific to just a given distro may be specify that distro as well.

Comment (Ivanidzo4ka) [reply docs/building/unix-instructions.md]: what about libiomp5 ? Mkl need it as well

Comment (artidoro) [reply docs/building/unix-instructions.md]: I just checked and both are installed because libomp5 is a dependency for libomp-dev.

Comment (Ivanidzo4ka) [reply docs/building/unix-instructions.md]: Good to know.

**Pull request merged**